File: CS201/Assignments/PythonJavaAll/commandline.py
'''
   commandline.py
   Jeff Ondich, 2013-01-04
   Anna Rafferty, 2016-08-24 (conversion to python3)
   Layla Oepser, 2017-08-30


   Identical to linereader.py, but takes the input file path from
   the command line rather than from a hard-coded string. Look at
   the comments for the java version for more about command line
   arguments.
   
   To run, type the following at the command line:
   "python3 linereader.py somefile.txt"
   
   Intended as the Python half of parallel examples in Python and
   Java. See CommandLine.java.
'''

# START CHANGES
# This is the part that's different from linereader.py.
import sys

if len(sys.argv) != 2:
    print('Usage: python %s inputFilePath' % sys.argv[0], file=sys.stderr)
    exit()
    
inputFilePath = sys.argv[1]
#END CHANGES

try:
    inputFile = open(inputFilePath)
except Exception as e:
    print('Cannot open', inputFilePath, sys.stderr)
    exit()

numberOfLines = 0
for line in inputFile:
    print(line.upper(), end='')
    numberOfLines += 1
print('\nNumber of lines in file:', numberOfLines)

File: CS201/Assignments/PythonJavaAll/CommandLine.java
import java.io.File;
import java.util.Scanner;
import java.io.FileNotFoundException;

/**
 * CommandLine.java
 * Jeff Ondich, 2014-01-05
 * Modified: Anna Rafferty, 2016-08-24
 * Modified: Layla Oesper, 2017-08-30
 *
 * A copy of LineReader.java that takes the input file path from the command
 * line rather than from a hard-coded string. For example, I might run this like:
 * java CommandLine somefile.txt
 * "somefile.txt" is a command line argument (an argument entered in terminal)
 * and the program will read this and use it as the filename to read from.
 * Note that you've seen command line arguments in other contexts - for example,
 * you've probably typed something likd "ls Documents" to list all the files
 * in the directory named Documents.
 * 
 * Once you've tried running this program, try modifying it to so that if there
 * is more than one command line argument, it prints out the second one.
 * 
 * This is the Java half of a pair of parallel examples in Python and Java.
 * See commandline.py.
 */
public class CommandLine {
    public static void main(String[] args) {
        // START CHANGES
        // This is the only part that's different from LineReader.java
        if (args.length == 0) {
            System.err.println("Usage: java CommandLine inputFilePath");
            System.exit(1);
        }
        String inputFilePath = args[0];
        // END CHANGES

        File inputFile = new File(inputFilePath);

        Scanner scanner = null;
        try {
            scanner = new Scanner(inputFile);
        } catch (FileNotFoundException e) {
            System.err.println(e);
            System.exit(1);
        }

        int numberOfLines = 0;
        while (scanner.hasNextLine()) {
            String line = scanner.nextLine();
            System.out.println(line.toUpperCase());
            numberOfLines++;
        }
        scanner.close();
        System.out.println("\nNumber of lines in file: " + numberOfLines);
    }
}
